llvm/include/llvm/CAS/CASDB.h

```diff
@@ -248,6 +248,8 @@ class CASDB : public CASIDContext {
   template <class ProxyT, class HandleT>
   Expected<ProxyT> loadObjectProxy(Expected<HandleT> H);
 
+  virtual Error validateID(const CASID &ID) = 0;
```

Comment:
How about changing this to `validateObject`? I think validating an "ID" is unclear that it's validating the whole object not just the ID itself.